upup/pkg/fi/cloudup/azure/azure_cloud.go

```diff
@@ -128,9 +134,34 @@ func (c *azureCloudImplementation) AddClusterTags(tags map[string]*string) {
 }
 
 func (c *azureCloudImplementation) GetApiIngressStatus(cluster *kops.Cluster) ([]kops.ApiIngressStatus, error) {
-	// TODO(kenji): Implement this. Currently we return nil as we
-	// don't create any resources for ingress to the API server.
-	return nil, nil
+	var ingresses []kops.ApiIngressStatus
+
+	// Get scale sets in cluster resource group and find masters scale set
+	scaleSets, err := c.vmscaleSetsClient.List(context.TODO(), cluster.Spec.CloudConfig.Azure.ResourceGroupName)
+	if err != nil {
+		return nil, fmt.Errorf("error getting Cluster Master Scale Set for API Ingress Status: %s", err)
+	}
+	var vmssName string
+	for i := range scaleSets {
+		if strings.Contains(*scaleSets[i].Name, ".masters.") {
+			vmssName = *scaleSets[i].Name
+			break
+		}
+	}
+
+	// Get masters scale set network interfaces and append to api ingress status
+	nis, err := c.NetworkInterface().ListScaleSetsNetworkInterfaces(context.TODO(), cluster.Spec.CloudConfig.Azure.ResourceGroupName, vmssName)
+	if err != nil {
+		return nil, fmt.Errorf("error getting Master Scale Set Network Interfaces for API Ingress Status: %s", err)
+	}
+	for _, ni := range nis {
+		for _, i := range *ni.IPConfigurations {
+			ingresses = append(ingresses, kops.ApiIngressStatus{
+				IP: *i.PrivateIPAddress,
+			})
+		}
```

Comment:
I'm wondering how we should handle VMs that have multiple network interfaces or multiple IPConfigurations. I _think_ we would want to only have one ApiIngressStatus for each master VM. Would there be situations in which a master VM might have multiple? Perhaps a CNI provider that adds NIs or IPs to VMs, for example. Openstack has some additional metadata it uses to look for a specific "fixed" IP on the instance, does Azure support anything similar?

https://github.com/kubernetes/kops/blob/55e38c96d3b83c7dd7c89c205fb0a1f2914e7851/upup/pkg/fi/cloudup/openstack/utils.go#L111